### PR TITLE
fix!: Update min provider to TPG ~> 3.44, CI fixes

### LINKED ISSUE #88: Appears xip.io shutdown 2021-May, current CI depends on xip.io
### TL;DR

Appears xip.io shutdown 2021-May - the current CI depends on xip.io

nip.io might be a drop-in replacement

### Expected behavior

_No response_

### Observed behavior

_No response_

### Terraform Configuration

```hcl
N/A
```

### Terraform Version

```sh
N/A
```

### Additional information

_No response_

### LINKED ISSUE #92: google_sql_database_instance added gitlab_deletion_protection in TPG v3.44
### TL;DR

google_sql_database_instance added gitlab_deletion_protection in TPG v3.44

### Terraform Resources

_No response_

### Detailed design

_No response_

### Additional information

_No response_

### LINKED ISSUE #93: Add gitlab_db_random_prefix=true to simple_example and test to enable repeat local testing
### TL;DR

Add gitlab_db_random_prefix=true to simple_example and test to enable repeat local testing without week db name cool off,

### Terraform Resources

_No response_

### Detailed design

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by apeabody**
Closes #88 
Closes #92 
Closes #93

### CODE REVIEW

**Comment by comment-bot-dev** [comment]
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

File: Makefile [hunk 1]
```diff
@@ -18,7 +18,7 @@
 # Make will use bash instead of sh
 SHELL := /usr/bin/env bash
 
-DOCKER_TAG_VERSION_DEVELOPER_TOOLS := 1.0
+DOCKER_TAG_VERSION_DEVELOPER_TOOLS := 0.14
```
Comment (bharathkkb) [inline Makefile:21]: Why are we downgrading versions?

File: versions.tf [hunk 2]
```diff
@@ -15,15 +15,15 @@
  */
 
 terraform {
-  required_version = ">= 0.13.0"
+  required_version = ">= 0.13.0, < 15.0"
```
Comment (bharathkkb) [inline versions.tf:18]: This is a breaking change and same question as above w.r.t versions.
```suggestion
  required_version = ">= 0.13.0, < 15.0"
```

File: examples/simple_example/variables.tf [hunk 3]
```diff
@@ -17,3 +17,13 @@
...
+
+variable "gitlab_db_random_prefix" {
+  description = "Sets random suffix at the end of the Cloud SQL instance name."
+  default     = false
+}
+
+variable "gitlab_deletion_protection" {
+  description = "Must be false to allow Terraform to destroy the Cloud SQL instance."
+  default     = true
+}
```
Comment (bharathkkb) [inline examples/simple_example/variables.tf:29]: We can hardcode these in the examples rather than exposing vars.

**Review by bharathkkb — CHANGES_REQUESTED** [review]
Thanks @apeabody

File: Makefile (hunk 1 above)
Comment (apeabody) [reply Makefile:21]: The module and CI don't currently work with TF v0.15+. To support TF v0.15+ and TPG 4+ will be a longer/larger change, but if you prefer we can can wait on this commit and combine all the changes in a few days.

File: versions.tf (hunk 2 above)
Comment (apeabody) [reply versions.tf:18]: Same as above note.

File: Makefile (hunk 1 above)
Comment (bharathkkb) [reply Makefile:21]: In that case lets skip constraining the version in the module and just target this as a CI fix. Curious what happens with  TF v0.15+ that is blocking?

Comment (apeabody) [reply Makefile:21]: A major addition with TF v0.15 is sensitive output values, so that change is my initial suspect for what is impacting the k8s auth during the CI.

**Review by bharathkkb — APPROVED** [review]
